k9mail/src/main/java/com/fsck/k9/mailstore/ProvidedTempFileBody.java

```diff
@@ -82,11 +88,11 @@ public long getSize() {
         throw new IllegalStateException("Data must be fully written before it can be read!");
     }
 
-    public File getFile() throws IOException {
+    public Uri getProviderUri(String mimeType) throws IOException {
```

Comment:
This mixes two concepts. I think it's better if this class only knows about files. Constructing the content URI should be the responsibility of another class.